src/lib.rs

```diff
@@ -1,10 +1,47 @@
-use anyhow;
 use rev_buf_reader::RevBufReader;
-use std::io::BufRead;
+use std::io::{BufRead, BufReader};
 use std::{fmt, fs::File, io, path::Path};
 
 use chrono::NaiveDateTime;
 
+pub struct Seeker {
```

Comment:
I fancied trying to make the internal logic look like a [Builder pattern](https://rust-unofficial.github.io/patterns/patterns/creational/builder.html) so thats what i've tried to do here.